src/components/PrivateRoute.js

```diff
@@ -23,20 +24,23 @@ const PrivateRoute = ({children, ...rest}) => {
             }
         }
         fetchUser()
-    },[])
+     },[])
 
     if(isLoading){
```

Comment:
Don't return in multiple places. You are returning at line #29 and #32. This pattern might bring more confusion. What you can instead do is set the result to a variable. And than return that in the same place ( in the end ).